dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hive/AbstractHiveConnector.java

```diff
@@ -64,21 +61,25 @@ public abstract class AbstractHiveConnector extends AbstractConnector {
     @ThreadSafe
     public static class ThriftClientPool implements AutoCloseable {
 
+        public interface ThriftClientConsumer {
+            void accept(HiveMetastoreThriftClient thriftClient) throws Exception;
+        }
+
         @Nonnull
         private final String name;
         @Nonnull
         private final ThreadLocal<? extends HiveMetastoreThriftClient> threadLocalThriftClient;
         @Nonnull
-        private final ExecutorService executorService;
+        private final ExecutorManager executorManager;
         @Nonnull
         private final Object lock = new Object();
         @GuardedBy("lock")
         @Nonnull
         private final List<@NonNull HiveMetastoreThriftClient> builtClients = new ArrayList<>();
 
-        public ThriftClientPool(@Nonnull String name, @Nonnull HiveMetastoreThriftClient.Builder thriftClientBuilder, @Nonnull ExecutorService executorService) {
+        public ThriftClientPool(@Nonnull String name, @Nonnull HiveMetastoreThriftClient.Builder thriftClientBuilder, @Nonnull ExecutorManager executorManager) {
             this.name = Preconditions.checkNotNull(name, "name was null.");
-            this.executorService = Preconditions.checkNotNull(executorService, "executorService was null.");
+            this.executorManager = Preconditions.checkNotNull(executorManager, "executorManager was null.");
```

Comment:
This should create an ExecutorManager and keep it private. The caller doesn't need to know about it.